File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        hashp={}
    

        for i in range(len(arr)):
            temp=arr[i]*2
            div=arr[i]/2
            if temp in hashp or div in hashp:
                print(hashp)
                print(arr[i])
                return True
            else:
                hashp[arr[i]]=temp
                
   
        return False

        

File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {

        HashSet<Integer> box = new HashSet<>();

        for (int i : arr) {
            if ((box.contains(i / 2) && i % 2 == 0) || box.contains(i * 2)) {

                return true;
            } else {

                box.add(i);
            }

        }
        return false;
    }
}